src/widget/wpixmapstore.cpp

```diff
@@ -14,32 +14,30 @@ PaintablePointer WPixmapStore::getPaintable(const PixmapSource& source,
     if (source.isEmpty()) {
         return PaintablePointer();
     }
-    QString key = source.getId() + QString::number(mode) + QString::number(scaleFactor);
+    QString key = QStringLiteral("%1%2%3").arg(source.getId(),
+            QString::number(mode),
+            QString::number(scaleFactor));
 
     // See if we have a cached value for the pixmap.
-    PaintablePointer pPaintable = m_paintableCache.value(
-            key,
-            PaintablePointer());
-    if (pPaintable) {
-        return pPaintable;
+    auto it = m_paintableCache.find(key);
+    if (it != m_paintableCache.end()) {
+        return it.value();
     }
 
-    pPaintable = PaintablePointer(new Paintable(source, mode, scaleFactor));
-
+    PaintablePointer pPaintable(new Paintable(source, mode, scaleFactor));
     m_paintableCache.insert(key, pPaintable);
     return pPaintable;
 }
 
 // static
-QPixmap* WPixmapStore::getPixmapNoCache(
+std::unique_ptr<QPixmap> WPixmapStore::getPixmapNoCache(
         const QString& fileName,
         double scaleFactor) {
-    QPixmap* pPixmap = nullptr;
-    QImage* img = m_loader->getImage(fileName, scaleFactor);
-    pPixmap = new QPixmap();
-    pPixmap->convertFromImage(*img);
-    delete img;
-    return pPixmap;
+    auto* pImage = m_loader->getImage(fileName, scaleFactor);
+    if (!pImage || pImage->isNull()) {
+        return nullptr;
+    }
+    return std::make_unique<QPixmap>(QPixmap::fromImage(*pImage));
```

Comment:
```suggestion
    return std::unique_ptr(&QPixmap::fromImage(*pImage));
```
Otherwise `make_unique` will call the copy ctor with the temporary.

Reply:
whoops, blunder v2... 
Unless we eliminate the `unique_ptr` entirely, we need the copy to move it from the stack to the heap... We can think about removing it in a follow up PR (possibly replacing it with a std::optional instead).

Reply:
Reverted